tasks/message/message.go

```diff
@@ -278,13 +279,31 @@ func (p *MessageProcessor) extractMessageModels(ctx context.Context, ts *types.T
 			}
 			result.Messages = append(result.Messages, msg)
 
+			msgsSeen[message.Cid()] = struct{}{}
+
 			if p.parseMessages {
 				dstAddr, err := address.NewFromString(msg.To)
 				if err != nil {
 					return nil, nil, xerrors.Errorf("parse to address: %w", err)
 				}
 
-				st, err := state.LoadStateTree(p.node.Store(), ts.ParentState())
+				child, err := p.node.ChainGetTipSetByHeight(ctx, ts.Height()+1, types.NewTipSetKey())
+				if err != nil {
+					// If we aren't finalized, we fail for now, because a child tipset may occur
+					if head, err := p.node.ChainGetTipSet(ctx, types.NewTipSetKey()); err == nil && head.Height()-ts.Height() < 900 {
```

Comment:
```suggestion
					if head, err := p.node.ChainGetTipSet(ctx, types.NewTipSetKey()); err == nil && head.Height()-ts.Height() < build.Finality {
```
Prefer to avoid magic numbers. This const indirects to abi.ChainEpoch(900). https://github.com/filecoin-project/lotus/blob/master/build/params_shared_vals.go#L38

Reply:
What does passing in a Zero TipSetKey do for `ChainGetTipSet`? Is that a secret way to get ChainHead? And if so, why not `api.ChainHead()`?

Reply:
they call exactly the same `getHeaviestTipset` method in the implementation.